core/src/frame_lifecycle.rs

```diff
@@ -83,70 +82,41 @@ pub fn run_all_phases_avm2(context: &mut UpdateContext<'_, '_>) {
     let stage = context.stage;
 
     *context.frame_phase = FramePhase::Enter;
-
-    stage.enter_frame(context);
-
-    let mut ran_framescript = FnvHashSet::default();
-
     Avm2::each_orphan_movie(context, |movie, context| {
-        if movie.initialized() {
-            // Orphan frame scripts run in a really weird place.
-            // Running them here matches the output we get from Flash Player,
-            // where the currentFrame field field hasn't been updated yet
-            // (but is updated when we call an enterFrame listener)
-            movie.run_frame_scripts(context);
-            movie.enter_frame(context);
-            ran_framescript.insert(movie.downgrade().as_ptr());
-        }
+        movie.enter_frame(context);
     });
+    stage.enter_frame(context);
 
     *context.frame_phase = FramePhase::Construct;
-
-    Avm2::each_orphan_movie(context, |movie, context| {
-        if !movie.initialized() {
-            movie.run_frame_scripts(context);
-            ran_framescript.insert(movie.downgrade().as_ptr());
-        }
-    });
-
-    stage.construct_frame(context);
-
     Avm2::each_orphan_movie(context, |movie, context| {
         movie.construct_frame(context);
-        if !ran_framescript.contains(&movie.downgrade().as_ptr()) {
-            movie.run_frame_scripts(context);
-        }
     });
-
+    stage.construct_frame(context);
     stage.frame_constructed(context);
 
     *context.frame_phase = FramePhase::Update;
-    stage.run_frame_avm2(context);
-
     Avm2::each_orphan_movie(context, |movie, context| {
         movie.run_frame_avm2(context);
     });
+    stage.run_frame_avm2(context);
```

Comment:
Is the change in ordering between orphans and the stage correct?

Reply:
This method doesn't actually do anything, and will be removed in https://github.com/ruffle-rs/ruffle/pull/9944